client/SharedComponents/withClickTracker.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import axios from 'axios';
+
+const withClickTracker = WrappedComponent => {
+  class WithClickTracker extends React.Component {
+    handleClickTrack(event) {
+      const postBody = {
+        element: event.target.toString(),
+        widget: WrappedComponent.name,
+        time: new Date().toString()
+      }
+
+      axios
+        .post('/interactions', postBody)
+        .then(results => console.log(results))
```

Comment:
Is this working for you? On the backend I found returning the console.log resulted in a unresolved promise because it was returning the results of the console.log, instead of the results. It was a little frustrating for me to test that way.

Reply:
This is the very end of the promise chain, so I don't believe it matters if the final then block returns anything. In any case, it is working on my end, but if you run into errors trying to implement it, let me know.